### DISCUSSION TITLE
Making an entity navigate a complex path

### DISCUSSION THREAD

**Discussion by AstroSquared** [Minestom/Minestom#1826] [Help] [answered]
I need a pathfinding system where an entity has to follow a path that has a lot of turns. The entity reaches the first point, then goes towards the second, and so on. I already have an array of points.

I've managed to do this in Spigot with the citizens api, but now to figure out how to do this in Minestom.

I can't find a method to tell an entity to follow a list of points, or a way to detect if the entity has finished pathfinding to give it another point.

Is there a way to do this in minestom?

**Comment by emortaldev** [comment, marked as answer]
You can rapidly teleport the entity to simulate movement.
You can lerp (linear interpolate) between the points in the array until you have reached the end while also tracking the distance between points to make sure it is moving at a constant speed.

**Reply by AstroSquared** [reply]
pretty sure i fixed this another way but this way would definitely work
